analyses/molecular-subtyping-HGG/09-HGG-with-braf-clustering.Rmd

```diff
@@ -0,0 +1,417 @@
+---
+title: "HGAT samples without histone mutations that have `BRAF V600E` mutations"
+output: 
+  html_notebook:
+    toc: TRUE
+    toc_float: TRUE
+author: Chante Bethell for ALSF CCDL
+date: 2020
+---
+
+This notebook will look at HGAT samples without histone mutations that have `BRAF V600E` mutations.
+
+The purpose is to identify samples that may need to be reclassified as LGAT instead of HGAT using t-SNE and UMAP clustering.
+
+## Usage
+
+This notebook is intended to be run via the command line from the top directory
+of the repository as follows:
+
+```
+Rscript -e "rmarkdown::render('analyses/molecular-subtyping-HGG/09-HGG-with-braf-clustering.Rmd', clean = TRUE)"
+```
+
+## Set up
+
+### Libraries and functions
+
+```{r}
+# Load in tidyverse functions
+library(tidyverse)
+```
+
+### Directories
+
+```{r}
+# File path to scratch directory -- this contains the cleaned data
+scratch_dir <- file.path("..", "..", "scratch")
+
+# File path to data directory
+data_dir <- file.path("..", "..", "data")
+
+# File path to plots directory -- this contains the multipanel output plots
+plots_dir <- file.path("plots")
+
+if (!dir.exists(plots_dir)) {
+  dir.create(plots_dir)
+}
+
+```
+
+### Functions
+
+```{r}
+# Source the functions for dimension reduction
+source(
+  file.path(
+    "..",
+    "transcriptomic-dimension-reduction",
+    "util",
+    "dimension-reduction-functions.R"
+  )
+)
+```
+
+#### Function to join metadata with HGG molecular subtype file
+
+```{r}
+align_data <- function(aligned_scores_df,
+                       hgg_molecular_subtype_df) {
+  # Given a data.frame with dimension reduction scores already aligned with the
+  # metadata, join the `molecular_subtype` column from the 
+  # `HGG_molecular_subtype.tsv` file generated in this module.
+  # Args:
+  #   aligned_scores_df: data.frame containing dimension reduction scores
+  #                      and relative information from the metadata
+  #   hgg_molecular_subtype_df: data.frame containing a `molecular_subtype`
+  #                             column with values determined in this module
+  # Returns:
+  #   aligned_scores_df: data.frame containing dimension reduction scores,
+  #                      metadata, and the `molecular_subtype` column from the
+  #                      `results/HGG_molecular_subtype.tsv` file
+  
+  aligned_scores_df <- aligned_scores_df %>%
+    dplyr::left_join(hgg_molecular_subtype_df, by = "sample_id") %>%
+    dplyr::group_by(sample_id) %>%
+    dplyr::select(-molecular_subtype.x, molecular_subtype = molecular_subtype.y) %>%
+    dplyr::distinct() %>%
+    dplyr::filter(
+      broad_histology %in% c(
+        "Low-grade astrocytic tumor",
+        "Diffuse astrocytic and oligodendroglial tumor"
+      )
+    )
+
+  return(aligned_scores_df)
+}
+```
+
+#### Function to generate plot list from dimension reduction scores
+
+```{r}
+# Alter the function to plot
+plot_dimension_reduction <- function(umap_aligned_scores_df,
```

Comment:
Could you instead run the `plot_dimension_reduction` function that you source from `analyses/transcriptomic-dimension-reduction/util/dimension-reduction-functions.R` twice - once with UMAP, once with t-SNE?

Reply:
I would need to adapt that function to account for the second variable that we would like to represent on the plots (molecular subtype in addition to broad histology). Should I change the function in the `transcriptomic-dimension-reduction` module in this PR? If so, I would likely need to propagate the changes downstream.

Reply:
The idea behind the issue is to specifically highlight the HGG samples that are H3 wildtype with BRAF V600E mutations - you can create a new variable that captures this information. You essentially have 3 groups you're interested in: HGAT samples that don't have `BRAF V600E` as a molecular subset, HGAT samples that _do_ have `BRAF V600E` as the molecular subset, and LGAT samples.

Reply:
Ahhh, got it 👍